apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/adapter/AdapterCommon.kt

```diff
@@ -77,7 +77,7 @@ internal fun readFromResponseCodeBlock(
     CodeBlock.of(
         "$T $L = $L;",
         context.resolver.resolveIrType(property.info.type),
-        context.layout.variableName(property.info.responseName),
+        property.responseName(context),
```

Comment:
`responseName` has a big "GraphQL" connotation, I'd avoid using it for kotlin names. 

For some general context about what I was trying to achieve with `CodegenLayout`, `CodegenLayout` takes GraphQL names as input and outputs "Target" names where "Target" is either Java or Kotlin.

In that specific case, I'd put the check for `__typename` in `CodegenLayout.variableName` itself. I don't know if that makes sense?

Reply:
I was hesitating about that 😅 Yeah thanks for the context it makes sense! Ok to do it in `variableName` then.